### PR TITLE
Anim autobuild

### PR DESCRIPTION
**Pull request by Rockett8855**
Added a `./genvids.py` built on `manim`, with some simple caching. See readme for usage.

### CODE REVIEW

File: animations/linear_regression/lr2_anim.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
+import numpy as np
```
Comment (hschafer) [inline animations/linear_regression/lr2_anim.py:1]: Is this the original animation file before I did the rewrite (e.g., adding the util classes and other animations)? I think you can delete this if it's stale.

File: animations/mlmodel_anim.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        self.play(FadeIn(fe))
+        self.play(ShowCreation(fe2mlm))
+        self.play(FadeIn(mlm))
+        self.play(ShowCreation(mlm2mla))
+        self.play(FadeIn(mla))
+        self.play(ShowCreation(qm2mla))
+        self.play(FadeIn(qm))
+
+    def train_data(self):
+        label = BTextMobject("Training \\\\ Data")
```
Comment (hschafer) [inline animations/mlmodel_anim.py:74]: You can use `r''` to make "raw strings" so you don't have to escape the backslashes.
```suggestion
        label = BTextMobject(r"Training \\ Data")
```

File: animations/mlmodel_anim.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        self.play(ShowCreation(td2qm2))
+        self.play(FadeIn(fe))
+        self.play(ShowCreation(fe2mlm))
+        self.play(FadeIn(mlm))
+        self.play(ShowCreation(mlm2mla))
+        self.play(FadeIn(mla))
+        self.play(ShowCreation(qm2mla))
+        self.play(FadeIn(qm))
+
+    def train_data(self):
```
Comment (hschafer) [inline animations/mlmodel_anim.py:73]: Is it possible to abstract these into one function like
```python
def make_box(self, text, box_color, text_color=color):
    label = BTextMobject(text, color=text_color)
    bg = Rectangle(color=box_color, fill_color=box_color, fill_opacity=0.7)
    bg.surround(label, buff=1.0)
    return VGroup(bg, label)
```

**Comment by hschafer** [comment]
The only thing that's a "nice to have" (can be part of this PR or another) is a way to copy these to the website dir.

I propose we assume that there is a similar file structure in the website for each animation. For example, the animation for `animations/linear_regression/example_anim.py` should always go to the `website` directory to the path `public/animations/linear_regression/example_anim.mp4`. 

Then whenever I make a bunch of changes (maybe to the core util files), I could just run this script and it will copy the results to the website for me. Maybe we want a `--copy` flag to indicate it should copy the files over.

File: animations/linear_regression/lr2_anim.py (hunk 1 above)
Comment (Rockett8855) [reply animations/linear_regression/lr2_anim.py:1]: sorry yeah lr_anim.py and lr2_anim.py are just placeholders, we can just delete them and integrate your actual changed in another pr

**Comment by Rockett8855** [comment]
> Did you mean to commit this `videos` directory with images?

`manim` automatically outputs image to wherever the video directory is, so this was just the simplest, it'll be easier to leave it this way unless you have a specfic objection, are we going to want to compile final images out to the `public` website directory enough to make adding that functionality worth it?

> The only thing that's a "nice to have" (can be part of this PR or another) is a way to copy these to the website dir.
> 
> I propose we assume that there is a similar file structure in the website for each animation. For example, the animation for `animations/linear_regression/example_anim.py` should always go to the `website` directory to the path `public/animations/linear_regression/example_anim.mp4`.
> 
> Then whenever I make a bunch of changes (maybe to the core util files), I could just run this script and it will copy the results to the website for me. Maybe we want a `--copy` flag to indicate it should copy the files over.

yes this can be done fairly easily. i'll add the copy flag. are you only going to want to copy over high quality videos, right now if you choose low quality or high quality it'll just append an _hq.mp4/_lq.mp4 to the file: i.e. `lr2_anim.py -> lr2_anim_hq.mp4 OR lr2_anim_lq.mp4.

**Comment by hschafer** [comment]
> > Did you mean to commit this `videos` directory with images?
> 
> `manim` automatically outputs image to wherever the video directory is, so this was just the simplest, it'll be easier to leave it this way unless you have a specfic objection, are we going to want to compile final images out to the `public` website directory enough to make adding that functionality worth it?
> 
> > The only thing that's a "nice to have" (can be part of this PR or another) is a way to copy these to the website dir.
> > I propose we assume that there is a similar file structure in the website for each animation. For example, the animation for `animations/linear_regression/example_anim.py` should always go to the `website` directory to the path `public/animations/linear_regression/example_anim.mp4`.
> > Then whenever I make a bunch of changes (maybe to the core util files), I could just run this script and it will copy the results to the website for me. Maybe we want a `--copy` flag to indicate it should copy the files over.
> 
> yes this can be done fairly easily. i'll add the copy flag. are you only going to want to copy over high quality videos, right now if you choose low quality or high quality it'll just append an _hq.mp4/_lq.mp4 to the file: i.e. `lr2_anim.py -> lr2_anim_hq.mp4 OR lr2_anim_lq.mp4.

* Sorry, I meant can we just ignore the `videos` directory entirely in `git`?  There really isn't a reason I want to version control any images from these animations (only a few times do I actually care about the final image, and if I do I just save it manually since it hasn't been super common). 
* And yeah, if you just copy the high quality right now that's perfect and we can build out more flexibility in quality if we feel like we need it later.

P.S. Why the hell does this not have threaded replies...

**Comment by Rockett8855** [comment]
> P.S. Why the hell does this not have threaded replies...

microsoft just wants you to use jira............................................................................

File: genvids.py [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+    DESCRIPTION = """
+        Generate animations for 'manim', will generate videos for any python
+        file with the suffix *_anim.py. Only the `Animation` class will be
+        compiled per individual filesh.
+
+        For any files in the animation tree, the output will be in the directory
+        specified by --out with the same directory structure, called *_anim.mp4,
+        in accordance with the python file.
+    """
+    parser = argparse.ArgumentParser(description=DESCRIPTION)
```
Comment (hschafer) [inline genvids.py:169]: Nit (but will merge anyway without a fix): A few common conventions in Python scripts
* Have an explicit `main` method that this `if` statement calls
* Have a method to build the `ArgumentParser` to get those low-level details out of main.

**Pull request merged**